packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -351,23 +529,105 @@ class _ZoomExitTransition extends StatelessWidget {
 
-  @override
-  Widget build(BuildContext context) {
-    final Animation<double> fadeTransition = reverse
-      ? _fadeOutTransition.animate(animation)
+  void _updateAnimations() {
+    fadeTransition = widget.reverse
+      ? _fadeOutTransition.animate(widget.animation)
       : kAlwaysCompleteAnimation;
-    final Animation<double> scaleTransition = (reverse
+    scaleTransition = (widget.reverse
       ? _scaleDownTransition
       : _scaleUpTransition
-    ).animate(animation);
+    ).animate(widget.animation);
+
+    widget.animation.addListener(_onAnimationValueChange);
+    widget.animation.addStatusListener(_onAnimationStatusChange);
+  }
+
+  @override
+  void initState() {
+    _updateAnimations();
+    delegate = _ZoomExitTransitionDelegate(
+      reverse: widget.reverse,
+      fade: fadeTransition,
+      scale: scaleTransition,
+    );
+    super.initState();
+  }
+
+  @override
+  void didUpdateWidget(covariant _ZoomExitTransition oldWidget) {
+    if (oldWidget.reverse != widget.reverse || oldWidget.animation != widget.animation) {
+      oldWidget.animation.removeListener(_onAnimationValueChange);
+      oldWidget.animation.removeStatusListener(_onAnimationStatusChange);
+      _updateAnimations();
+      delegate.dispose();
+      delegate = _ZoomExitTransitionDelegate(
+        reverse: widget.reverse,
+        fade: fadeTransition,
+        scale: scaleTransition,
+      );
+    }
+    super.didUpdateWidget(oldWidget);
+  }
 
-    return FadeTransition(
-      opacity: fadeTransition,
-      child: ScaleTransition(scale: scaleTransition, child: child),
+  @override
+  void dispose() {
+    widget.animation.removeListener(_onAnimationValueChange);
+    widget.animation.removeStatusListener(_onAnimationStatusChange);
+    delegate.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (allowRasterization) {
+      return RasterWidget(
+        delegate: delegate,
+        rasterize: rasterize,
+        child: widget.child,
+      );
+    }
+    return ScaleTransition(
+      scale: scaleTransition,
+      child: FadeTransition(
+        opacity: fadeTransition,
+        child: widget.child,
+      ),
     );
   }
 }
 
+class _ZoomExitTransitionDelegate extends RasterWidgetDelegate {
+  _ZoomExitTransitionDelegate({
+    required this.reverse,
+    required this.scale,
+    required this.fade,
+  }) {
+    scale.addListener(notifyListeners);
+    fade.addListener(notifyListeners);
+  }
+
+  final bool reverse;
+  final Animation<double> scale;
+  final Animation<double> fade;
+
+  @override
+  void paint(PaintingContext context, Rect area, ui.Image image, double pixelRatio) {
+    _drawImageScaledAndCentered(context, image, scale.value, fade.value, pixelRatio);
+  }
+
+  @override
+  bool shouldRepaint(covariant _ZoomExitTransitionDelegate oldDelegate) {
+    return oldDelegate.reverse != reverse || oldDelegate.fade != fade || oldDelegate.scale != scale;
```

Comment:
Technically, you'd only have to repaint if `oldDelegate.fade.value != fade.value` (and same for scale), right?

Reply:
Done